packages/@aws-cdk/aws-iam/lib/policy-document.ts

```diff
@@ -90,12 +91,21 @@ export class AccountPrincipal extends ArnPrincipal {
  * An IAM principal that represents an AWS service (i.e. sqs.amazonaws.com).
  */
 export class ServicePrincipal extends PolicyPrincipal {
-  constructor(public readonly service: string) {
+  /**
+   * @param scope   a Construct scoping the service principal, typically the resource that will cause the service
+   *                principal to perform actions this policy statement authorizes, or it's parent Construct.
+   * @param service the short name of the service (e.g: s3, sns, sqs, secretsmanager, ...)
+   */
+  constructor(private readonly scope: cdk.IConstruct, private readonly service: string) {
     super();
   }
 
   public policyFragment(): PrincipalPolicyFragment {
-    return new PrincipalPolicyFragment({ Service: [ this.service ] });
+    return new PrincipalPolicyFragment({
+      Service: [
+        Region.named(this.scope.node.stack.region).servicePrincipal(this.service)
```

Comment:
`stack.region` might return a Token. What will we do in those cases?

Reply:
That's the _hardest_ part. Sidelining this for now (probably would add a check+throws)... But could inject a mapping, or four use a context resolver...